codegen/src/main/twirl/templates/PlayScala/Router.scala.txt

```diff
@@ -15,13 +15,17 @@ import akka.stream.Materializer
 
 import play.api.mvc._
 import play.api.mvc.akkahttp.AkkaHttpHandler
-import play.api.routing.Router
+import play.api.routing.{Router, SimpleRouter}
 import play.api.routing.Router.Routes
 
 @@Singleton
-class @{service.name}Router @@Inject()(impl: controllers.@{service.name}Impl)(implicit mat: Materializer)
-  extends Router {
-  protected val prefix = @{service.name}.name
+class @{service.name}Router(impl: controllers.@{service.name}Impl, prefix: String)(implicit mat: Materializer)
+  extends SimpleRouter {
+  require(prefix != null && !prefix.isEmpty && prefix.charAt(0) == '/')
```

Comment:
Being careful here. We should probably have these checks in Play routers too and mention them in the Play scaladocs.